### DISCUSSION TITLE
BUG: Phantom not respecting the full MWA specification

### DISCUSSION THREAD

**Discussion by josip-fundl** [phantom/docs#126] [General]
In the Mobile Wallet Adapter specification it has been explicitly mentioned to use the wallet addresses from the (mwa wallet session) _authToken_ to sign a transaction that was requested by the user.

For example, I have my mobile dApp dReader and Phantom mobile installed on my android.

1. I go to dReader and hit the "connect wallet" button which opens up Phantom and asks me to confirm the connection between the apps.
2. In Phantom I had wallet A selected as active so when I hit "confirm" I'll connect my wallet A to dReader. This will generate an _authToken_ for me which encapsulates permissions for the wallet A to connect with dReader
3. I then close the dReader app, open Phantom wallet and switch the active wallet to wallet B
4. I go back to dReader and hit the "Buy" button
5. Phantom wallet session will then open and ask me to sign a transaction but transaction simulation will fail.

Why? Because from dReader I've used the _authToken_ from my wallet A and requested to sign the transaction with it, but on Phantom wallet B is now set to active. I'm essentially trying to sign a transaction for the wallet A, with the wallet B.

Of course it's failing.

What should happen in reality is, if Phantom detects that _authToken_ has been used to prompt a signTransaction event, it should read the wallet address which was authorized with the _authToken_ and sign the transaction with it. Regardless of which wallet is currently actively selected on Phantom mobile.

A bit more on this MWA spec can be found here: https://solana-mobile.github.io/mobile-wallet-adapter/spec/spec.html#authorize-and-sign-transaction

**Comment by Michaelsulistio** [comment]
Echoing this issue and adding some more context in a [previous GH issue where this was discussed](https://github.com/solana-mobile/mobile-wallet-adapter/issues/471). In the link, you can see a repro video of this bug.

This is an important fix that greatly helps UX for users with multiple accounts use dApps seamlessly. Let me know if there are questions on the implementation of this fix! Happy to help.

**Comment by adamdelphantom** [comment]
Thank you for reporting @d-reader-josip! We are looking into a resolution.

**Reply by josip-fundl** [reply]
Sweet, thanks!

**Reply by josip-fundl** [reply]
@Michaelsulistio from Solana Mobile can help with the fix if need be, especially if you require some more input about the problem
